pkg/kubelet/cm/devicemanager/rank.go

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package devicemanager
+
+import (
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog"
+)
+
+const (
+	MAXCOST = 10000
+)
+
+var costs [][]int = costs_()
+
+func costs_() [][]int { // communication cost between gpus
+	return [][]int{
+		{0, 1, 10, 10, 100, 100, 100, 100},
```

Comment:
gpu topo info can be retrieved by:
github.com/NVIDIA/gpu-monitoring-tools/bindings/go/nvml